UnitySDK/Assets/ML-Agents/Scripts/Sensor/SensorComponent.cs

```diff
@@ -14,5 +14,23 @@ public abstract class SensorComponent : MonoBehaviour
         /// </summary>
         /// <returns></returns>
         public abstract ISensor CreateSensor();
+
+        /// <summary>
+        /// Returns the shape of the sensor observations that will be created.
+        /// </summary>
+        /// <returns></returns>
+        public abstract int[] GetObservationShape();
```

Comment:
Added this to the editor components; they now have to return the size that the sensors will create.